Replace parseAddress() with parseKeyFromAddress()

christianbundy opened the pull request:
`ssbRef.parseAddress()` is printing deprecation traces whenever it's
used, but it can be replaced with the simple `parseKeyFromAddress()`
instead.

christianbundy committed:
Replace parseAddress() with parseKeyFromAddress()

`ssbRef.parseAddress()` is printing deprecation traces whenever it's
used, but it can be replaced with the simple `parseKeyFromAddress()`
instead.

staltz commented on src/index.ts:
```diff
@@ -27,8 +27,8 @@ class LAN {
-    const peer = Ref.parseAddress(address);
-    if (peer?.key !== this.ssb.id) {
+    const peerKey = Ref.getKeyFromAddress(address);
+    if (peerKey !== this.ssb.id) {
```
could we do `peerKey && peerKey !== this.ssb.id`? Because `peer?.key` actually contained a check whether the parsing succeeded or not. But now, if `peerKey` is undefined, this if condition would pass.

christianbundy replied:
Sweet, thanks for the fix!

staltz commented:
Good catch!

christianbundy committed:
Fix handler when getKeyFromAddress() fails

The code in `master` uses `peer?.key`, which fails when `peer` is
undefined, so we should handle this the same way. The peer key is only
useful if it's truthy.

staltz commented:
Thanks!

staltz commented:
Release 0.1.4
